# Scrapping data from Ph Santos review videos

Libraries

In [144]:
# Basic Import
import json
import os
import time

from yt_dlp import YoutubeDL
from youtube_transcript_api import YouTubeTranscriptApi

## Getting all the videos ids

The following code uses yt-dlp cli to download all the video urls and their respective titles
into a single file called list4.txt.

The code is structured as:

1. --flat-playlist: list all the videos but don't download them
2. --match-title: search for titles that contains "Crítica"
3. --reject-title: search for titles that DOES NOT contains "série"
4. --skip-download: skips the download of videos that matches the reject-title flag
and the ones that DOES NOT match the match-title flag
5. --print-to-file: what are the informations that will be printed to the file. In this case
url and title
6. list4.txt: the name of the file that will have the 
7. last one in the url for the video list from the Youtube Channel

In [19]:
os.system('yt-dlp -s --flat-playlist --match-title "Crítica" --reject-title "série" --skip-download --print-to-file url,title list4.txt "https://www.youtube.com/@phsantos/videos"')

[youtube:tab] Extracting URL: https://www.youtube.com/@phsantos/videos
[youtube:tab] @phsantos/videos: Downloading webpage
[download] Downloading playlist: PH Santos - Videos
[youtube:tab] UCjY_LBR_7-Ile0MAsbeJWBg page 1: Downloading API JSON
[youtube:tab] UCjY_LBR_7-Ile0MAsbeJWBg page 2: Downloading API JSON
[youtube:tab] UCjY_LBR_7-Ile0MAsbeJWBg page 3: Downloading API JSON
[youtube:tab] UCjY_LBR_7-Ile0MAsbeJWBg page 4: Downloading API JSON
[youtube:tab] UCjY_LBR_7-Ile0MAsbeJWBg page 5: Downloading API JSON
[youtube:tab] UCjY_LBR_7-Ile0MAsbeJWBg page 6: Downloading API JSON
[youtube:tab] UCjY_LBR_7-Ile0MAsbeJWBg page 7: Downloading API JSON
[youtube:tab] UCjY_LBR_7-Ile0MAsbeJWBg page 8: Downloading API JSON
[youtube:tab] UCjY_LBR_7-Ile0MAsbeJWBg page 9: Downloading API JSON
[youtube:tab] UCjY_LBR_7-Ile0MAsbeJWBg page 10: Downloading API JSON
[youtube:tab] UCjY_LBR_7-Ile0MAsbeJWBg page 11: Downloading API JSON
[youtube:tab] UCjY_LBR_7-Ile0MAsbeJWBg page 12: Downloading API JSON
[youtu

0

In [52]:
with open('list4.txt', 'r') as file:
    lista_video_id = file.read().split('\n')
    lista_video_id = list(map(lambda x: x.split('?v=')[-1], lista_video_id ))
    lista_video_id = lista_video_id[0::2]

In [131]:
criticas_list = []

for critica_id in lista_video_id:
    critica_dict = {}

    try: 
        transcricao_sep = YouTubeTranscriptApi.get_transcript(critica_id, languages=['pt'])
        transcricao_junto = ''

        for frase in transcricao_sep:
            transcricao_junto = transcricao_junto + ' ' + frase['text'].replace('[Música]', '')
        
        critica_dict['id'] = critica_id
        critica_dict['texto'] = transcricao_junto
        
        criticas_list.append(critica_dict)

    except Exception as e:
        print(critica_id, e)

    

1_OkQTJ3Hng no element found: line 1, column 0
uFXzaklwQRw no element found: line 1, column 0
4hXUixvNwv8 no element found: line 1, column 0
4I6ZeYlwocw 
Could not retrieve a transcript for the video https://www.youtube.com/watch?v=4I6ZeYlwocw! This is most likely caused by:

No transcripts were found for any of the requested language codes: ['pt']

For this video (4I6ZeYlwocw) transcripts are available in the following languages:

(MANUALLY CREATED)
None

(GENERATED)
 - nl ("Dutch (auto-generated)")[TRANSLATABLE]

(TRANSLATION LANGUAGES)
 - ab ("Abkhazian")
 - aa ("Afar")
 - af ("Afrikaans")
 - ak ("Akan")
 - sq ("Albanian")
 - am ("Amharic")
 - ar ("Arabic")
 - hy ("Armenian")
 - as ("Assamese")
 - ay ("Aymara")
 - az ("Azerbaijani")
 - bn ("Bangla")
 - ba ("Bashkir")
 - eu ("Basque")
 - be ("Belarusian")
 - bho ("Bhojpuri")
 - bs ("Bosnian")
 - br ("Breton")
 - bg ("Bulgarian")
 - my ("Burmese")
 - ca ("Catalan")
 - ceb ("Cebuano")
 - zh-Hans ("Chinese (Simplified)")
 - zh-Hant ("Ch

In [134]:
len(criticas_list)

653

In [135]:
with open('criticas.txt', 'w') as file:
    file.write(json.dumps(criticas_list, ensure_ascii=False))

## Basic Statistic from the dataset

- Dataset size: 653 reviews
- Oldest Review: 2016-05-13
- Newest Review: 2024-11-01

In [148]:
n_criticas = len(criticas_list)
print('Número de críticas:', n_criticas)

Número de críticas: 653


In [151]:
tamanho_criticas = [len(criticas_list[i]['texto'].split(' ')) for i in range(0, n_criticas)]

653